### DISCUSSION TITLE
address with port 80 already binded after postal installation. How to enable link click tracking by setting up the virtual hosts then?

### DISCUSSION THREAD

**Discussion by wonderbee77** [postalserver/postal#2307] [Help with using Postal]
How to enable link click tracking by setting up the virtual hosts? any detailed document? After postal installation, I am now trying to create a virtual host with a subdomain. But getting the below error.

Dec 29 07:00:46 postal.jio-labs.org apachectl[485173]: (98)Address already in use: AH00072: make_sock: could not bind to address [::]:80
Dec 29 07:00:46 postal.jio-labs.org apachectl[485173]: (98)Address already in use: AH00072: make_sock: could not bind to address 0.0.0.0:80

**Comment by wonderbee77** [comment]
Caddy is running in port 80. That is the reason. By any chance do you have any documentation on adding virtual hosts to caddy server?

**Reply by willpower232** [reply]
traditionally you'll be running one web server on both ports 80 and 443, running both apache and caddy at the same time is difficult and they naturally can't share the ports.

You'll need to either include the tracking links in the caddy config or include the postal forwarding in the apache config. It comes down to whichever you're most comfortable doing.

**Comment by wonderbee77** [comment]
thanks...

```
postal.jio-labs.org {
  reverse_proxy 127.0.0.1:5000
}

 click.jio-labs.org {
   reverse_proxy 127.0.0.1:8092 {
     header_up X-Postal-Track-Host "1"
   }
 }
```

are the above CaddyFile configs right? I do have a postal mail server installed in postal.jio-labs.org. and another simple web app running on port 8092. I have a CNAME record to point click.jio-labs.org to track.postal.jio-labs.org. On the other hand, I have an A record to point track.postal.jio-labs.org to the IP address of the machine.

still, click tracking is not working. I have random links in the body of the email and clicking it.

**Reply by willpower232** [reply]
Presumably you need to `reverse_proxy 127.0.0.1:5000` in all cases you expect it to go to postal? so proxying click.jio-labs.org to your simple web app isn't right?

**Comment by wonderbee77** [comment]
any thoughts?
